include/fastrtps/utils/collections/ResourceLimitedVector.hpp

```diff
@@ -119,6 +119,46 @@ class ResourceLimitedVector
         return *this;
     }
 
+    /**
+     * Insert value before pos.
+     *
+     * @param pos   iterator before which the content will be inserted. pos may be the end() iterator.
+     * @param value element value to insert.
+     *
+     * @return Iterator pointing to the inserted value. end() if insertion couldn't be done due to collection limits.
+     */
+    iterator insert(
+            const_iterator pos,
+            const value_type& value)
+    {
+        if (capacity() >= configuration_.maximum)
+        {
+            return end();
+        }
+
+        return collection_.insert(pos, value);
+    }
+
+    /**
+     * Insert value before pos.
+     *
+     * @param pos   iterator before which the content will be inserted. pos may be the end() iterator.
+     * @param value element value to insert.
+     *
+     * @return Iterator pointing to the inserted value. end() if insertion couldn't be done due to collection limits.
+     */
+    iterator insert(
+            const_iterator pos,
+            value_type&& value)
+    {
+        if (capacity() >= configuration_.maximum)
+        {
+            return end();
+        }
+
+        return collection_.insert(pos, value);
```

Comment:
value is now a left value.
```suggestion
        return collection_.insert(std::move(pos), value);
```

Reply:
I guess you mean `collection_.insert(pos, std::move(value));`

Reply:
exactly 😅